### ISSUE TITLE
Policy for glossary entries

### ISSUE THREAD

**Issue by dpritchLibre** [BarkleyBG/LCC#81]
I've been trying to think about what a sensible approach to including glossary entries is for the last couple of days and I'd like to ask for your input about this at some point.  There are two issues to think about as I see it.

The main question that I have is what entries should be included in the glossary.  When I first tried to include terms I started by including everything, even terms such as `file`, `Microsoft Windows`, and so on.  But after some rumination I think this is kind of ridiculous.  First it is a big waste of our time writing entries for things that people already know about, and secondly it bloats the glossary and useful definitions are overwhelmed by unnecessary ones.  So after some thought / experimentation I thought that maybe a reasonable policy would be the following: to include a term whenever we feel the necessity to explain / define it in the main text.  I think that this policy produced reasonable results in the revisions for Chapter 1, which you can see in one of my branches / PRs.

The second question is what policy to adopt as far as referring to glossary entries.  One policy would be to refer to an entry every single time it appears in the text.  An alternative policy that I think may work well is to refer to the entry only in sections where the item is defined or added to.  I've listed some advantages / disadvantages to both approaches below:

Refer to glossary entry every single time it appears in the text:
*  advantages: super easy for reader to link to glossary definition and then back to text
*  disadvantages: makes the source code super ugly / annoying to read, I find that having so many clickeable words kind of annoying / distracting as the reader, and most importantly I think it makes the glossary less valuable b/c the links to the text that it provides then become pretty useless

Refer to glossary entry only in sections where the item is defined or added to:
*  advantages: cleaner source code, less clickeable links may be less distracting to read, makes the glossary more valuable b/c the reader can use it to find the section(s) where a term is defined / described
*  disadvantages: inconvenient for the reader to navigate to the glossary, reader doesn't know whether term is in glossary or not by waving mouse over it

---------------------------------------------------------------------

So anyway I'd like to hear your thoughts on these topics when time permits.  I'm pretty flexible on what we do but I think we should make some decisions so that they are consistent throughout the document and we can add them to `CONTRIBUTING.md`.

In summary the two questions I think we should answer are:
1.  Policy for what items to include in glossary
2.  Policy for when to link items from test to their glossary entry

**Comment by BarkleyBG** [comment]
Hey! Sorry for the delay. I haven't been getting any emails from GH, so I missed a few things. I'll try to get caught up. Hopefully my email settings work now.

I think that referring to the glossary item every time it is used may be overkill. In some scenarios, I think we use a phrase a dozen times on a page. Linking all of them is overkill. It might be easier to link only, say, the first reference in a chapter/section, but then it could be even harder for the writer to check whether it's been used before! So I'm unsure.

Can you make an edit to change one of these to a "disadvantage"?

> advantages: inconvenient for the reader to navigate to the glossary, reader doesn't know whether term is in glossary or not by waving mouse over it
    advantages: cleaner source code, less clickeable links may be less distracting to read, makes the glossary more valuable b/c the reader can use it to find the section(s) where a term is defined / described

**Comment by dpritchLibre** [comment]
Edited typo as per comment above

**Comment by BarkleyBG** [comment]
Email notifications are up and running, it seems.

**Comment by BarkleyBG** [comment]
I think that for now, since we're still developing/writing and not quite editing/improving, I'll suggest this rule of thumb:

1. If you wonder if a word ("*foo*") should have a glossary entry, then it should (`\gls{foo}`)
2. If you are writing the word _foo_ and you haven't written `\gls{foo}` in this current chapter (or section), then you should write `\gls{foo}` and not `foo`.
2. If you are writing the word _foo_ and you wrote `\gls{foo}` in the previous sentence, then it is not necessary to write `\gls{foo}` again, and using the non-linked `foo` is fine.
3. Similarly, if you wrote `\gls{foo}` in the same/previous paragraph, then it's probably not necessary to write `\gls{foo}` again, and using the non-linked `foo` is fine.
4. If you are writing the word _foo_ and you cannot recall the last time you wrote `\gls{foo}`, then you should write `\gls{foo}` and not `foo`.

Then, when finished writing, we can go back and edit to a consistent style. Like: (A) Always use the `\gls{}` link, or (B) Use the `\gls{}` link at the first reference of each section (and in other circumstances where it seems necessary)

**Comment by dpritchLibre** [comment]
Okay, I think I am good with this.  So when you say "section" do you mean as a literal `\section` as opposed to `\subsection` or `\subsubsection`?  I'm going to start following this for my revisions for chapter 2 if so.

For (1), I think a less subjective rule would be "to include a term whenever we feel the necessity to explain / define it in the main text."

**Comment by BarkleyBG** [comment]
Section should be sufficient but I'm open to ideas

I like your edit to (1)

**Comment by dpritchLibre** [comment]
Cool beans.  I'll try to add this to `CONTRIBUTING.md` at some point.

**Comment by BarkleyBG** [comment]
I'll PR this tomorrow

**Comment by BarkleyBG** [comment]
closed from #93
